app/src/main/kotlin/com/simplemobiletools/calculator/activities/CryptoActivity.kt

```diff
@@ -0,0 +1,149 @@
+
+        var numberPickerFROM: NumberPicker = numberPicker1
+        var numberPickerTO: NumberPicker = numberPicker2
+
+        // Values of each of the Number Pickers
+        val valuesFROM = arrayOf("Bitcoin", "Ethereum", "Ripple", "Litecoin", "USD")
+        val valuesTO = arrayOf("Ethereum", "Ripple", "Litecoin", "USD", "Bitcoin")
+
+        // Setting default value of each Number Picker
+        numberPickerFROM.minValue = 0
+        numberPickerTO.minValue = 0
+
+        // Setting size of each Number Picker
+        numberPickerFROM.maxValue = valuesFROM.size - 1
+        numberPickerTO.maxValue = valuesTO.size - 1
+
+        // Setting Number Pickers to the values
+        numberPickerFROM.displayedValues = valuesFROM
+        numberPickerTO.displayedValues = valuesTO
+
+        // Sets whether the selector wheel wraps when reaching the min/max value.
+        numberPickerFROM.wrapSelectorWheel = true
+        numberPickerTO.wrapSelectorWheel = true
+
+        val cryptoSymbolHashMap = HashMap<String, String>()
+        cryptoSymbolHashMap.put("Ethereum", "ETH")
+        cryptoSymbolHashMap.put("Bitcoin", "BTC")
+        cryptoSymbolHashMap.put("Ripple", "XRP")
+        cryptoSymbolHashMap.put("Litecoin", "LTC")
+        cryptoSymbolHashMap.put("USD", "USD")
+
+        btn_convert_cryto.setOnClickListener { _ ->
+            cryptoFROM = cryptoSymbolHashMap.get(valuesFROM[numberPickerFROM.value]).toString()
+            cryptoTO = cryptoSymbolHashMap.get(valuesTO[numberPickerTO.value]).toString()
+
+            if (cryptoTO == cryptoFROM) {
+                calc.overwriteNumber(Formatter.stringToDouble(getResult()))
+            } else {
+                performConversion(cryptoFROM, cryptoTO)
+            }
+        }
+        btn_delete.setOnClickListener { calc.handleDelete(); checkHaptic(it) }
+        btn_delete.setOnLongClickListener { calc.handleClear(); true }
+        result.setOnLongClickListener { copyToClipboard(result.value); true }
+
+        AutofitHelper.create(result)
+    }
+
+    private fun performConversion(cryptoFROM: String, cryptoTO: String) {
+        if (isOnline()) {
+            calc.calculateCrypto(cryptoFROM, cryptoTO, this)
+        } else if (!isOnline()) {
+            displayToast("It seems there has been a connection problem, contact you ISP for more details !")
+        }
+    }
+
+    fun isOnline(): Boolean {
+        val cm = getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        val netInfo = cm.activeNetworkInfo
+        return netInfo != null && netInfo.isConnectedOrConnecting
+    }
+
+    @SuppressLint("MissingSuperCall")
+    override fun onResume() {
+        super.onResume()
+        if (storedUseEnglish != config.useEnglish) {
+            restartActivity()
+            return
+        }
+
+        if (storedTextColor != config.textColor) {
+            updateViewColors(crypto_holder, config.textColor)
+//            updateButtonColor(config.customPrimaryColor)
```

Comment:
Is this necessary ?

Reply:
It is not. I shall remove that commented line, thank you for spotting it.

Reply:
c: